### PR TITLE
AnyStr cleanup

### PR DESCRIPTION
**Pull request by srittau**
* Replace all uses of StrPath, BytesPath, and AnyPath in Python 2 stubs.
* Add StrOrBytesPath as preferred alias for AnyPath.
* Replace all remaining AnyPath instances with StrOrBytesPath.
* Mark AnyPath as obsolete.

Part of #5470

### CODE REVIEW

File: stdlib/@python2/bz2.pyi [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import io
-from _typeshed import AnyPath, ReadableBuffer, WriteableBuffer
-from typing import IO, Any, Iterable, List, Optional, TypeVar, Union
+from _typeshed import ReadableBuffer, WriteableBuffer
+from typing import IO, Any, Iterable, List, Optional, Text, TypeVar, Union
```
Comment (Akuli) [inline stdlib/@python2/bz2.pyi:3]: Why not just use `unicode` instead of `Text` in Python 2?

Comment (JelleZijlstra) [reply stdlib/@python2/bz2.pyi:3]: I think we agreed at some point that semantically `Text` means str or unicode and `unicode` means just unicode.

File: stdlib/_typeshed/__init__.pyi [hunk 2]
```diff
@@ -58,11 +58,11 @@ class SupportsItemAccess(SupportsGetItem[_KT_contra, _VT], Protocol[_KT_contra,
...
     def __delitem__(self, __v: _KT_contra) -> None: ...
 
-# StrPath and AnyPath can be used in places where a
-# path can be used instead of a string, starting with Python 3.6.
+# These aliases are simple strings in Python 2.
 StrPath = Union[str, PathLike[str]]
 BytesPath = Union[bytes, PathLike[bytes]]
-AnyPath = Union[str, bytes, PathLike[str], PathLike[bytes]]
+StrOrBytesPath = Union[str, bytes, PathLike[str], PathLike[bytes]]
+AnyPath = StrOrBytesPath  # obsolete, will be removed soon
```
Comment (JelleZijlstra) [inline stdlib/_typeshed/__init__.pyi:65]: I thought you were planning to turn this into an AnyStr-like TypeVar

**Review by JelleZijlstra — APPROVED** [review]

File: stdlib/@python2/bz2.pyi (hunk 1 above)
Comment (srittau) [reply stdlib/@python2/bz2.pyi:3]: The information about this is all a bit hidden, but `unicode` and `str` are not compatible on all type checkers (e.g. pytype). Therefore the following can fail to type check:

```python
def foo(x: unicode): ...
foo("")
```

But PEP 484 also says:

> For parameters typed as `Text`, arguments of type `str` as well as `unicode` should be acceptable.

So the following checks on all type checkers:

```python
def foo(x: Text): ...
foo("")
```

File: stdlib/_typeshed/__init__.pyi (hunk 2 above)
Comment (srittau) [reply stdlib/_typeshed/__init__.pyi:65]: I was, but I wanted to give this a bit of deprecation, then a bit of removal, then a resurrection (with about one mypy release between each), just to give people a change to react.

Comment (srittau) [reply stdlib/_typeshed/__init__.pyi:65]: Also the next mypy release will cause lots of problems, and I want to avoid to add this one to it.

**Pull request merged**